### DISCUSSION TITLE
Is there a plan to release a TF 2.4.0 compatible Horovod Dockerfile?

### DISCUSSION THREAD

**Discussion by asahalyft** [horovod/horovod#2822] [Q&A]
Hello Team,

I see the official Docker file still uses TF2 2.3.0 https://github.com/horovod/horovod/blob/386be429b1417a1f6cb5e715bbe36efd2e74f402/Dockerfile.gpu#L4

Can we expect a TF2 2.4.x compatible Docker file release in the Horovod repo?

**Comment by tgaddair** [comment]
Hey @asahalyft, we'll definitely publish a Docker with the latest versions of the frameworks for the next release. Though one thing we might consider doing is publishing new images automatically through CI/CD with the latest master.

**Reply by asahalyft** [reply]
@tgaddair Yes, that would be great. However, I look at the https://github.com/horovod/horovod/blob/master/Dockerfile.gpu to get inspirations to create our own company specific custom horovod image.

Since, there was none so I took a stab and have managed to build a Dockerfile with the following:
```
ENV TENSORFLOW_VERSION=2.4.1
ENV PYTORCH_VERSION=1.7.0
ENV TORCHVISION_VERSION=0.8.1
ENV CUDNN_VERSION=8.0.4.30-1+cuda11.0
ENV NCCL_VERSION=2.8.4-1+cuda11.0
```

**Comment by pescap** [comment]
Hi! Is there a reason why you use TensorFlow 2.7.3 for the Docker image?
https://github.com/horovod/horovod/blob/e4146b54fa39010f8e1129d42226367173d3947f/docker/horovod/Dockerfile#L8

Do you plan to bump TF to a more recent version (e.g. 2.10.0)? 

Thank you! Else, I will build my custom image as explained [here](https://github.com/horovod/horovod/blob/a304c81f4c4816445ed736bf8f1d4b97403726ed/docker/README.md).

**Reply by maxhgerlach** [reply]
This is scheduled for the next release, see https://github.com/horovod/horovod/issues/3689
